packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,224 @@
+/// <reference lib="dom" />
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { TelegramCloudStorageStamperError } from "./errors";
+
+declare global {
+  interface Window {
+    Telegram: any;
+  }
+}
+
+// config when telegram stamper is passed in an api key
+export type TTelegramCloudStorageStamperConfig = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+  cloudStorageKey?: string;
+};
+
+type CloudStorageAPIKey = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+};
+
+// Constant for default key name
+const DEFAULT_TURNKEY_CLOUD_STORAGE_KEY = "TURNKEY_API_KEY";
+
+/**
+ * Stamper to use within a `TurnkeyClient`
+ */
+export default class TelegramCloudStorageStamper {
+  // This stamper uses a typical apikey stamper under the hood and abstracts away the storage of the actual api keys
+  stamper: ApiKeyStamper | undefined;
+  cloudStorageKey: string | undefined;
+
+  private constructor(config: TTelegramCloudStorageStamperConfig) {
+    // check to see if were in a telegram mini app context
+    TelegramCloudStorageStamper.checkTelegramContext();
+
+    // instantiate the stamper
+    this.stamper = new ApiKeyStamper({
+      apiPublicKey: config.apiPublicKey,
+      apiPrivateKey: config.apiPrivateKey,
+    });
+
+    // set the cloud storage key
+    this.cloudStorageKey = config.cloudStorageKey;
+  }
+
+  // create a telegram stamper by getting/setting the private/public api key values from/to telegram cloud storage
+  static async create(
+    config?: TTelegramCloudStorageStamperConfig
+  ): Promise<TelegramCloudStorageStamper> {
+    // check to see if were in a telegram mini app context
+    TelegramCloudStorageStamper.checkTelegramContext();
+
+    let cloudStorageKey = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY;
+    let apiPublicKey = "";
+    let apiPrivateKey = "";
+
+    // set the api pub and priv key if config is passed in
+    if (config) {
+      apiPublicKey = config.apiPublicKey;
+      apiPrivateKey = config.apiPrivateKey;
```

Comment:
if the API public / private key is null/undefined/empty we should probably throw here? Otherwise we're gonna fail with "TelegramCloudStorageStamperError(`Failed storing api key in Telegram Cloud Storage`)"

Reply:
Its not possible for them to be null/undefined/etc.. because the function takes an optional argument of type `TTelegramCloudStorageStamperConfig` which requires that those 2 fields are populated. If the config is not passed in, this code path is skipped

Reply:
So, maybe not `null` or `undefined`, but empty is still possible? (`""`)

That might be enough of an edge case to be "okay" though.

Reply:
Yeah, theoretically they can pass any value into there "abc" which wouldnt be a valid apikey. Was kinda relying on api-key-stamper's error handling there